vmdb/app/controllers/application_controller/filter.rb

```diff
@@ -392,15 +392,17 @@ def exp_changed
     end
 
     # Check for changes in date format
-    if params[:date_format_1]
-      @edit[@expkey][:val1][:date_format] = params[:date_format_1]
-      @edit[@expkey][:exp_value].collect!{|v| v = params[:date_format_1] == "s" ? nil : EXP_TODAY}
-      @edit[@expkey][:val1][:through_choices] = exp_through_choices(@edit[@expkey][:exp_value][0]) if params[:date_format_1] == "r"
-    end
-    if params[:date_format_2]
-      @edit[@expkey][:val2][:date_format] = params[:date_format_2]
-      @edit[@expkey][:exp_cvalue].collect!{|v| v = params[:date_format_2] == "s" ? nil : EXP_TODAY}
-      @edit[@expkey][:val2][:through_choices] = exp_through_choices(@edit[@expkey][:exp_cvalue][0]) if params[:date_format_2] == "r"
+    if @edit[@expkey][:exp_value].present?
+      if params[:date_format_1]
+        @edit[@expkey][:val1][:date_format] = params[:date_format_1]
+        @edit[@expkey][:exp_value].collect!{|v| v = params[:date_format_1] == "s" ? nil : EXP_TODAY}
+        @edit[@expkey][:val1][:through_choices] = exp_through_choices(@edit[@expkey][:exp_value][0]) if params[:date_format_1] == "r"
+      end
+      if params[:date_format_2]
```

Comment:
@romanblanco : seems we should have 2 branches here and check for `exp_cvalue` in the 2nd half.